### PR TITLE
2 packages from bcpierce00/unison at 2.53.7

### PR DESCRIPTION
**Pull request by jhjourdan**
This pull-request concerns:
- `unison.2.53.7`: File-synchronization tool for Unix and Windows
- `unison-gui.2.53.7`: Pseudo-package for Unison GUI

---
* Homepage: https://www.cis.upenn.edu/~bcpierce/unison/
* Source repo: git://github.com/bcpierce00/unison.git
* Bug tracker: https://github.com/bcpierce00/unison/issues

---
:camel: Pull-request generated by opam-publish v2.3.0

### CODE REVIEW

**Comment by jhjourdan** [comment]
Is there a way to silence the lint about the missing dependency on Dune? There is a dune file, but the opam package file does not depend on it.

In addition, the package does not seem to work on Windows. I don't have a Windows machine to test and fix this, so is there a way to just tell that this package is not wompatible with Windows ?

**Comment by raphael-proust** [comment]
> is there a way to just tell that this package is not wompatible with Windows ?

You can add `available: os != "win32"` to the opam file.

File: packages/unison-gui/unison-gui.2.53.7/opam [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+separately, and then brought up to date by propagating the changes in
+each replica to the other."""
+url {
+  src:
+    "https://github.com/bcpierce00/unison/archive/refs/tags/v2.53.7.tar.gz"
+  checksum: [
+    "md5=e49a78c1e159d4d2315fb5264c83ed59"
+    "sha512=11bd1d2792fb84fa2b29426516d7b91a2295febcb84052118d492a29d9ddaa23265b66cc88113019ec6782f3edcf596f7a37c8637e673b2928188248b9d63d60"
+  ]
+}
```
Comment (raphael-proust) [inline packages/unison-gui/unison-gui.2.53.7/opam:31]:
```suggestion
}
available: os != "win32"
```

File: packages/unison/unison.2.53.7/opam [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+separately, and then brought up to date by propagating the changes in
+each replica to the other."""
+url {
+  src:
+    "https://github.com/bcpierce00/unison/archive/refs/tags/v2.53.7.tar.gz"
+  checksum: [
+    "md5=e49a78c1e159d4d2315fb5264c83ed59"
+    "sha512=11bd1d2792fb84fa2b29426516d7b91a2295febcb84052118d492a29d9ddaa23265b66cc88113019ec6782f3edcf596f7a37c8637e673b2928188248b9d63d60"
+  ]
+}
```
Comment (raphael-proust) [inline packages/unison/unison.2.53.7/opam:35]:
```suggestion
}
available: os != "win32"
```

**Comment by raphael-proust** [comment]
> Is there a way to silence the lint about the missing dependency on Dune?

I don't think that's possible but we can just ignore the error.

**Comment by raphael-proust** [comment]
Re: the windows failure, `pkg-config` has a different name on windows. I think you can fix it by changing the call you make on windows:
- call `pkgconf` (instead of `pkg-config`
- add the `--personality=x86_64-w64-mingw32` (or other personality value)

You can do that directly in the opam file:

```
build: [
  ["pkgconf" {os = "win32" & os-distribution = "cygwin"}
   "--personality=i686-w64-mingw32" {os = "win32" & os-distribution = "cygwin" & host-arch-x86_32:installed}
   "--personality=x86_64-w64-mingw32" {os = "win32" & os-distribution = "cygwin" & host-arch-x86_64:installed}
   "pkg-config" {os != "win32" | os-distribution != "cygwin"}
   REST OF THE PKGCONF ARGUMENTS HERE   
  ]
  REST OF THE BUILD INSTRUCTIONS HERE
]
```

You might be able to get away with a `$(PKGCONF)` variable if you are using a Makefile too.

But setting as unavailable on windows (as per the code suggestions above) is also ok.

**Comment by avsm** [comment]
> Is there a way to silence the lint about the missing dependency on Dune?

You can add `dune` to the `depopts` field instead. This indicates that the package will be recompiled if `dune` appears in the future, but is not required to install it without dune present.

**Comment by jhjourdan** [comment]
@raphael-proust Unison in itself does not directly depend on pkg-config. Instead, conf-gtk3 (and, I guess, lablgtk3) does.

Has anyone published a package using lablgtk3 working on Windows?

In the meantime, I pushed a version where Windows is disabled.

**Comment by jhjourdan** [comment]
The build failure is suprious:

```
fatal: unable to access 'https://github.com/ocaml/opam-repository.git/': Could not resolve host: github.com
```

File: packages/unison-gui/unison-gui.2.53.7/opam [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+maintainer: "unison-hackers@lists.seas.upenn.edu"
+authors: [
+  "Trevor Jim"
+  "Benjamin C. Pierce"
+  "Jérôme Vouillon"
+]
+license: "GPL-3.0-or-later"
+homepage: "https://www.cis.upenn.edu/~bcpierce/unison/"
+bug-reports: "https://github.com/bcpierce00/unison/issues"
+dev-repo: "git://github.com/bcpierce00/unison.git"
```
Comment (mseri) [inline packages/unison-gui/unison-gui.2.53.7/opam:11]: I think something is off, this package does not have `build` and `install` instructions, right?

Comment (jhjourdan) [reply packages/unison-gui/unison-gui.2.53.7/opam:11]: Indeed: it simply has an additional dependency on lablgtk. The makefile of the main package decides to build the GUI depending on whether lablgtk is alrady installed.

Comment (jhjourdan) [reply packages/unison-gui/unison-gui.2.53.7/opam:11]: That's the reason why lablgtk is a depopt of unison

Comment (mseri) [reply packages/unison-gui/unison-gui.2.53.7/opam:11]: Oh I see, so that indeed forces a recompilation. Thanks!

**Pull request merged**